src/drivers/intel/hda/hda-dma.c

```diff
@@ -28,6 +28,68 @@
 #include <stddef.h>
 #include <stdint.h>
 
+/*
+ * HD-Audio DMA programming sequence
+ *
+ * To start a stream (Playback):
+ * 1. Host sends the PCM_PARAMS IPC with two_step_stop flag set
+ * 2. Host sets RUN bit for link DMA
```

Comment:
what happens in the case of a mixer where the link DMA might already be running?
Also is there an order between 1 and 2? I would think not, but maybe the order is between 2 and 3?

Reply:
The ordering of 1, 2 and 3 is correct for pipelines with the mixer, isnt it? There's always a PCM_PARAMS before trigger start and thats when this flag would get set. And The ordering between 2 and 3 depends on the trigger order we set in the host. For playback, this order is correct. 

In the case of mixer, we send the PCM_PARAMS but since the link DMA is already running, 2 would be skipped. But, even in the FW, the PCM_PARAMS for the second pipeline are not propagated beyond the mixer to the DAI.

Reply:
maybe for the first stream it's valid, but when a second stream gets mixed we wouldn't start the link DMA?

Reply:
right, do you want me to mention this here? Maybe we need a separate section for mixer-based streams that we can add with the deep-buffer patches?

Reply:
yes, it would be good to mention here, and we'll revisit this with the deep-buffer cases...